### PR TITLE
[CenterOfMass]: Add bounding box

### PR DESCRIPTION
**Pull request by chupins**
Draw a bounding box around all selected object when you choose to look at the center of masses

### CODE REVIEW

**Comment by s-quirin** [comment]
@chupins We better should test it internally first and not confuse the user with numerous updates. Please check out [my fork](https://github.com/s-quirin/FreeCAD-macros) with Bugfixes.

**Comment by chupins** [comment]
@s-quirin, I merged your changes in [my fork](https://github.com/chupins/FreeCAD-macros).
And then, it's in the present merge request

File: Information/CenterOfMass.FCMacro
Comment (galou) [inline Information/CenterOfMass.FCMacro:273]: The original was nicer. Why this change?

File: Information/CenterOfMass.FCMacro
Comment (galou) [inline Information/CenterOfMass.FCMacro:397]: Wouldn't `Toggle` be better? In any case, the sentence should start with a capital letter.

File: Information/CenterOfMass.FCMacro
Comment (galou) [inline Information/CenterOfMass.FCMacro:625]: I see a strange sign in Github (`Â`). Is it normal?

File: Information/CenterOfMass.FCMacro
Comment (galou) [inline Information/CenterOfMass.FCMacro:800]: Please use single-quotes.

File: Information/CenterOfMass.FCMacro
Comment (galou) [inline Information/CenterOfMass.FCMacro:807]: Please activate or remove this line.

File: Information/CenterOfMass.FCMacro
Comment (galou) [inline Information/CenterOfMass.FCMacro:904]: This function doesn't use `self`, so it should be written as a free function.

**Review by galou — CHANGES_REQUESTED** [review]
Thanks for the changes! I didn't test them but just had a look at the code. There are a few things that can be done to improve it, please consider them. Thanks!

File: Information/CenterOfMass.FCMacro
Comment (chupins) [reply Information/CenterOfMass.FCMacro:273]: I have a strange (Â) that is appearing sometimes. The ^3 seems more "stable"

**Comment by chupins** [comment]
Sorry but I can't see anything above that clearly explains what's broken !
I don't know if it's the proper way to do it, but your MR contains a lot of thing that are already pending here.

**Comment by chupins** [comment]
A lot of changes made by s-quirin in #129 were already made here.
So I manually merged its version in here and took @galou remarks into account.

@s-quirin can you check if you find any problem ?
When I do a diff between your version and this one, I can't find any fundamental difference (the only things are my new colormaps and the mass/volume button)

File: Information/CenterOfMass.FCMacro
Comment (chupins) [reply Information/CenterOfMass.FCMacro:638]: It look like passing through different os the ³ changes in Â³

Comment (galou) [reply Information/CenterOfMass.FCMacro:638]: I'm sorry but I don't believe this. This is not a problem with the OS but with your editor. You should set the endocing to UTF-8 without BOM.

**Comment by galou** [comment]
@chupins: @s-quirin removed some files in his PR, can you do the same if these files are indeed not needed.

I'll wait for @s-quirin's approval before merging.

**Comment by galou** [comment]
@chupins: @s-quirin removed some files in his PR, can you do the same if these files are indeed not needed.

I'll wait for @s-quirin's approval before merging.

File: Information/CenterOfMass.FCMacro
Comment (chupins) [reply Information/CenterOfMass.FCMacro:638]: You're right, it's the text editor ! (but some time I swtich between windows and linux and I also switch between text editors...)

**Comment by chupins** [comment]
You're right again, I forgot to delete these unecessary files.
We'll see with @s-quirin how we merge the PR and the #129.

**Comment by galou** [comment]
@chupins @s-quirin I don't know whether I should merge this PR, #129, or both. Can you please express yourself here? Thanks!

**Comment by s-quirin** [comment]
> @chupins @s-quirin I don't know whether I should merge this PR, #129, or both. Can you please express yourself here? Thanks!

We couldn't come to an agreement yet because of holidays / lack of time. #128 mainly contains two controversial changes that emerged in my own test and with end users:
- The originally worked out color palette was replaced by a pre-selection from matplotlib. I revived the built-in colormap and added a setting for the user to add any colormap to the pre-selection.
- A feature in which the visualization is not about the mass but about the volume. The user can confuse both. A clear UI solution is not easy and should be postponed to a next version.

I asked @chupins to fix them in this PR or to withdraw it and go with #129.

**Comment by chupins** [comment]
@s-quirin has added everything on #129 .
We prefer canceling here and ask if #129 to be accepted.

**Pull request closed without merging**
